daras_ai_v2/search_ref.py

```diff
@@ -297,8 +304,62 @@ def render_output_with_refs(state, height=500):
         html = render_text_with_refs(text, state.get("references", []))
         scrollable_html(html, height=height)
 
-
 FOOTNOTE_SYMBOLS = ["*", "†", "‡", "§", "¶", "#", "♠", "♥", "♦", "♣", "✠", "☮", "☯", "✡"]  # fmt: skip
 def generate_footnote_symbol(idx: int) -> str:
     quotient, remainder = divmod(idx, len(FOOTNOTE_SYMBOLS))
     return FOOTNOTE_SYMBOLS[remainder] * (quotient + 1)
+
+def find_alphanumeric_window(s, window_size, reverse=False):
```

Comment:
I'll definitely say Bravo! for figuring this out. It seems to work pretty well. 

But at the same time it is a bit complex and prone to failures + is compute inefficient because of the n x m string window loops.

https://developer.mozilla.org/en-US/docs/Web/URI/Fragment/Text_fragments

I did some digging and realised you don't need both start and end strings, and can only do the start string. on top of that, you can provide multiple of these strings. It feels like we can build redundancy using this method, and can do it entirely in regex and avoid slower python loops given the requirement as per sean is to only be able to scroll to the relevant section, not necessarily highlight everything.


E.g. for this chunk:

```
TV Movie

Werewolf by Night (art)

Web Series

WHiH Newsfront (footage)

Docuseries

Legends
*The Avengers Initiative

Game

Avengers: Damage Control

Comic

Iron Man 3 Prelude
Thor: The Dark World Prelude
Captain America: Homecoming (mentioned)
The Avengers Adaptation
Avengers: Age of Ultron Prelude - This Scepter'd Isle
Avengers: Operation HYDRA
Captain America: Civil War Prelude Infinite Comic
Captain America: Road to War
Spider-Man: Homecoming Prelude
Avengers: Infinity War Prelude
Ant-Man and the Wasp Prelude
Captain Marvel Prelude
Avengers: Endgame Prelude
Spider-Man: Far From Home Prelude
Black Widow Prelude (flashbacks)

Book

Avengers: Infinity War: The Heroes' Journey
Captain Marvel: What Makes A Hero
The Wakanda Files: A Technological Exploration of the Avengers and
Beyond
Look Out for the Little Guy! (mentioned)

"The Avengers. That's what we call ourselves. We're sort of like a team.
'Earth's Mightiest Heroes'-type thing."

―Tony Stark to Loki^([src])

The Avengers were a team of extraordinary individuals, with either
superpowers or other special characteristics. Though primarily
affiliated with the interests of the United States of America, the
group's purpose was to protect global stability from inner or
extraterrestrial threats. The Avengers were first assembled by
S.H.I.E.L.D. as a result of the Avengers Initiative, when Loki invaded
Earth with his Chitauri army. The team, consisting of Iron Man, Captain
America, Hulk, Thor, Black Widow and Hawkeye defeated Loki and went
their separate ways for a while.
```

current code gives: https://marvelcinematicuniverse.fandom.com/wiki/Avengers#:~:text=Captain%20America%3A%20Civil%20War,separate%20ways%20for%20a%20while


But if we do the other method of using multiple sentences (joined together using `&text=`) then it sorta works:

https://marvelcinematicuniverse.fandom.com/wiki/Avengers#:~:text=Avengers:%20Age%20of%20Ultron%20Prelude%20-%20This%20Scepter'd%20Isle&text=Captain%20Marvel%20Prelude&text=The%20Wakanda%20Files:%20A%20Technological%20Exploration%20of%20the%20Avengers%20and&text=The%20Avengers%20were%20a%20team%20of%20extraordinary%20individuals&text=Look%20Out%20for%20the%20Little%20Guy!&text=The%20Avengers%20were%20a%20team%20of%20extraordinary%20individuals